src/app/components/marketplace/marketplace.component.ts

```diff
@@ -0,0 +1,172 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { BehaviorSubject, EMPTY, Observable, catchError, map, switchMap, tap } from 'rxjs';
+import { Resource, ServiceCard } from '../../types/dtos';
+import { MarketplaceService } from '../../services/marketplace.service';
+import { QueryService } from 'src/app/services/query.service';
+import { MatGridListModule } from '@angular/material/grid-list';
+import { CommonModule } from '@angular/common';
+import { MatCardModule } from '@angular/material/card';
+import { MatSelectChange, MatSelectModule } from '@angular/material/select';
+import { MatDividerModule } from '@angular/material/divider';
+import { MatIconModule } from '@angular/material/icon';
+import { HttpErrorResponse } from '@angular/common/http';
+
+export interface CarouselIndexes {
+    dataResourceIndex: number;
+    serviceAccessPointIndex: number;
+    physicalResourceIndex: number;
+    groupId?: number;
+}
+
+enum IndexType {
+    DataResource = 'dataResourceIndex',
+    ServiceAccessPoint = 'serviceAccessPointIndex',
+    PhysicalResource = 'physicalResourceIndex',
+}
+
+@Component({
+    selector: 'app-marketplace',
+    templateUrl: './marketplace.component.html',
+    standalone: true,
+    imports: [CommonModule, MatCardModule, MatGridListModule, MatSelectModule, MatDividerModule, MatIconModule],
+    styleUrls: ['./marketplace.component.scss'],
+})
+export class MarketplaceComponent implements OnInit {
+    legalNames$: Observable<string[]> = EMPTY;
+    services$: Observable<ServiceCard[]> = EMPTY;
+    legalName: string | null = null;
+    error: HttpErrorResponse | null = null;
+
+    private legalNameSubject = new BehaviorSubject<string | null>(null);
+    private legalName$ = this.legalNameSubject.asObservable();
+
+    public IndexType = IndexType;
+    public serviceIndexes: Map<number, CarouselIndexes> = new Map<number, CarouselIndexes>();
```

Comment:
The bookkeeping on the service indexes is pretty hard to follow. It seems to work fine, but it would be much simpler if we had a dedicated component for each service. It could take care of fetching its own service data (which are already separate requests).